roles/repository/tasks/main.yml

```diff
@@ -143,8 +142,17 @@
     dest: "{{ settings_folder }}/classpath/alfresco/web-extension/{{ item }}"
     mode: 'u=rwx,g=rwx,o=rx'
   loop: "{{ share_templates }}"
-  changed_when: false
 
+- name: Add Hazelcast cluster config for Share
+  template:
+    owner: "{{ username }}"
+    group: "{{ group_name }}"
+    src: custom-slingshot-application-context.xml
+    dest: "{{ settings_folder }}/classpath/alfresco/web-extension/custom-slingshot-application-context.xml"
+    mode: 'u=rw,g=,o='
+  become: true
+  when: groups['repository'] | length > 1 and not cluster_keepoff
```

Comment:
when there is only one node it's using an embedded hazelcast configuration?

in addition, to be truly idempotent the file needs to be absent when length <= 1

Reply:
yes, kind of, absence of this file disable share clustering which is the only  purpose of hazelcast.
Regarding the second part of your comment, I think that's already what the `when:`condition` does...?

Reply:
AFAIK `when` is a global parameter that runs or not the referenced module, so I don't think it's like having `state: absent` when `when: false`

googling doesn't help me, can you try to investigate too?

a fast solution that comes to my mind is to have another task with file module with a `when: ! $previous_task_condition` :D

Reply:
:-/ I won't comment on the proposed solution XD
What's the use-case you're trying to cover? The case where someone wants to move away from cluster and return to a single node? Or something else?

Reply:
yeah, I think we can ignore this but was quite surprised isn't something we can easily support